web/components/Feed/Feed.tsx

```diff
@@ -299,18 +341,36 @@ function Feed(props: {
         } else {
           tempState[filt.key].push(filt.options[ind]);
         }
-        console.log(tempState);
-        return tempState;
+        toReturn = tempState;
+        break;
       default:
-        return state;
+        toReturn = state;
     }
+    setQueryFilters(getQueryFilters(toReturn));
+    return toReturn;
   }
 
+  const [queryFilters, setQueryFilters] = useState(
+    getQueryFilters(getInitialFilters())
+  );
+
   const [selectedFilters, setSelectedFilters] = useReducer(
     filterReducer,
     getInitialFilters()
   );
 
+  const feedPosts = trpc.post.getFilteredPosts
+    .useQuery(queryFilters)
+    ?.data?.map((p: IPost) => {
+      return {
+        attachments: p.attachments,
+        date: p.date,
+        name: p.name,
+        tag: p.type,
+        description: p.description,
+      };
+    });
```

Comment:
Instead of mapping over it on the frontend, consider changing the output of the API to your liking